wls-gui-wahllokalsystem/src/components/ergebnisermittlung/TheErgebnisermittlungStimmzettelumschlaegeCard.vue

```diff
@@ -0,0 +1,77 @@
+<template>
+  <v-container>
+    <v-card>
+      <v-card-title>Zählen der Stimmzettel</v-card-title>
+      <v-card-text class="pb-0 pt-2">
+        <v-form
+          ref="anzahlStimmzettelForm"
+          v-model="anzahlStimmzettelValidForm"
+        >
+          <div class="d-flex flex-wrap justify-start">
+            <div>
+              <v-number-input
+                v-model="wahl.stimmzettelumschlaege.anzahlWaehler"
+                class="mr-4"
+                :rules="[REQUIRED, MIN_NUMBER(0), MAX_NUMBER(9999)]"
+                min-width="30rem"
+                data-test="numberInputAnzahlStimmzettel"
+                label="Anzahl der Stimmzettel"
+                clearable
+              />
+            </div>
+          </div>
+        </v-form>
+      </v-card-text>
+      <v-card-actions>
+        <base-button-save
+          active
+          :loading="isStimmzettelumschlaegeSaving"
+          :disabled="isSaveButtonDisabled"
+          @click="onSaveAnzahlStimmzettelClicked"
+        />
+      </v-card-actions>
+    </v-card>
+  </v-container>
+</template>
+
+<script setup lang="ts">
+import { computed, ref } from "vue";
+import {
+  VCard,
+  VCardActions,
+  VCardText,
+  VCardTitle,
+  VContainer,
+  VForm,
+  VNumberInput,
+} from "vuetify/components";
+
+import BaseButtonSave from "@/components/common/buttons/BaseButtonSave.vue";
+import { useWahlenStore } from "@/stores/wahlenStore.ts";
+import { MAX_NUMBER, MIN_NUMBER, REQUIRED } from "@/util/rules.ts";
+
+const props = defineProps<{
+  wahlId: string;
+}>();
+
+const {
+  getWahlOrUndefinedById,
+  saveStimmzettelumschlaege,
+  isStimmzettelumschlaegeSaving,
+} = useWahlenStore();
+
+const wahl = getWahlOrUndefinedById(props.wahlId);
+if (!wahl) {
+  throw new Error(`Wahl with ID ${props.wahlId} not found`);
+}
```

Comment:
ein if einfach so im skript aufzurufen fühlt sich für mich falsch an, ich finde das sollte immer in funktionen sein deswegen würde ich es in onMounted packen. ich würde aber eigentlich keinen fehler werfen, sondern eine toasty meldung draus machen, bzw noch besser wäre es glaube ich, das auf die view auszulagern die erst noch kommt, weil dann der menüpunkt garnicht erst angezeigt werden sollte, wenn es keine wahl mit der id gibt
```suggestion
onMounted(() => {
  if (!wahl) {
    throw new Error(`Wahl with ID ${props.wahlId} not found`);
    );
  }
});
```

Reply:
Ich hatte erst gar kein Error Handling drin und darauf verwiesen, dass diese Komponente aus der View ausschliesslich mit gültiger wahlID aufgerufen wird. CodeRabbit hat sich aber so sehr beschwert, dass ich diesen Vorschlag aufgenommen habe, so kommt halt eine leere Seite und ein Fehlerkommentar im Log.
Ich nehme es aber mal in die onMounted() rein, finde ich auch schöner.